api-public/src/main/kotlin/gropius/graphql/GropiusGraphQLContextFactory.kt

```diff
@@ -1,34 +1,51 @@
 package gropius.graphql
 
 import com.expediagroup.graphql.server.spring.execution.DefaultSpringGraphQLContextFactory
+import gropius.GropiusPublicApiConfigurationProperties
 import gropius.authorization.GropiusAuthorizationContext
+import gropius.model.user.GropiusUser
 import gropius.model.user.permission.NodePermission
 import gropius.repository.user.GropiusUserRepository
 import io.github.graphglue.authorization.AuthorizationContext
 import io.github.graphglue.authorization.Permission
+import io.jsonwebtoken.JwtException
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.io.Decoders
+import io.jsonwebtoken.security.Keys
 import kotlinx.coroutines.reactor.awaitSingle
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.server.ServerRequest
 
 /**
  * Generates the GraphQL context map
- * TODO: use authentication as soon as available
  *
  * @param gropiusUserRepository used to get the user
+ * @param gropiusPublicApiConfigurationProperties used to determine if authentication is optional
  */
 @Component
 class GropiusGraphQLContextFactory(
-    private val gropiusUserRepository: GropiusUserRepository
+    private val gropiusUserRepository: GropiusUserRepository,
+    private val gropiusPublicApiConfigurationProperties: GropiusPublicApiConfigurationProperties
 ) : DefaultSpringGraphQLContextFactory() {
 
+    /**
+     * Jwt parser based on the secret defined by [gropiusPublicApiConfigurationProperties]
+     */
+    private val jwtParser = Jwts.parserBuilder()
+        .setSigningKey(Keys.hmacShaKeyFor(Decoders.BASE64.decode(gropiusPublicApiConfigurationProperties.jwtSecret)))
+        .build()
+
     override suspend fun generateContextMap(request: ServerRequest): Map<*, Any> {
-        //TODO use authentication as soon as available
-        val userId = request.headers().firstHeader("Authorization")
-        val additionalContextEntries = if (userId == null) {
-            emptyMap()
+        val token = request.headers().firstHeader("Authorization")
+        val additionalContextEntries = if (token == null) {
+            if (gropiusPublicApiConfigurationProperties.debugNoAuthentication) {
+                emptyMap()
```

Comment:
Shouldn't debug mode add admin for every request?

Reply:
efac170f5511a19759d36954aa57d29c2ce9e6a2  
I forgot the necessary code a few lines below  
Regarding your question, it depends. Without a token, no authorization context is put into the map. By the query part, this is interpreted as no authorization required, therefore allowing all READ access to everything.
Mutations however require a token, as mutations need to know who performs the mutation (at least in a lot of cases).
The fix mentioned by the commit above fixes this behavior, as now debugNoAuthentication behaves like isAdmin for the authenticated user